### ISSUE TITLE
RFE: Drop native, outdated fedora plugins.

### ISSUE THREAD

**Issue by leamas** [OpenCPN/plugins#28]
During the initial development I had some native fedora plugins available. IMHO, in the long run this doesn't make much sense. Fedora is moving too fast, two major releases every year and dropping support for anything older than one year.

Fedora ha recognized this, and been one of the major stakeholders in the flatpak project. The way to support Fedora is finishing the flatpak release and plugins. As a bonus, this works on basically all Linux platforms.

The current Fedora plugins are at unsupported F29 which illustrates the problem.

While Fedora is my primary platform, I still think flatpak is the way to go and that we should drop the Fedora plugins like we dropped the rpm support for main opencpn.

**Comment by bdbcat** [comment]
Alec...
I agree, strategically.
But I wonder, do not average users complain about the size of flapak images?

**Comment by leamas** [comment]
The initial download is huge, agreed. But it's a initial thing, installing the environment. opencpn flatpak updates weighs in  at about 35Mb, nothing spectacular. And if user already is using flatpak'd apps chance is that the environment is already in place.

Also note that the downloaded environment is what makes flatpak run on multiple os:es and versions. It cannot really be avoided.

Bottom line: the initial download is a drawback, but nothing blocking. Many users download charts which are substantially bigger.

**Comment by bdbcat** [comment]
OK, I'll play with flatpak to get familiar with the UX.
btw, is oeSENC ok on flatpak now?

**Comment by leamas** [comment]
>OK, I'll play with flatpak to get familiar with the UX.

I have rebuilt master and published on the provisionary beta site. To use:

    $ flatpak remote-add --user --no-gpg-verify ocpn-beta http://opencpn.duckdns.org/opencpn-beta/website/repo/
    $ flatpak install ocpn-beta org.opencpn.OpenCPN

> btw, is oeSENC ok on flatpak now?

Still struggle. The decryption process seems to work, takes several hours, no messages about missing libraries and the log logs OK. But no charts displayed.

**Comment by leamas** [comment]
> Still struggle. The decryption process seems to work....

... but when I load the charts I find a number of entries like this in the log. Does it mean that the fimngerprint validation fails somehow? Any hint?

```
06:13:23 CET: validateUserKey
06:13:23 CET: Current userKey: r5Bl-PTSO-Jk1v-KEQq-BhGi-HA
06:13:23 CET: validateUserKey E1
06:13:23 CET:    Cannot load SENC file /home/mk/Charts/oesenc.sv/flatpak-SE-2020-7/OC-46-ZXUIE5.oesenc
06:13:23 CET: Retry...
06:13:23 CET: Available FALSE, retry... nLoop: 1
06:13:23 CET: Available FALSE, retry... nLoop: 2
06:13:24 CET: Available FALSE, retry... nLoop: 3
06:13:24 CET: Available FALSE, retry... nLoop: 4
06:13:25 CET: Available FALSE, retry... nLoop: 5
06:13:25 CET: Available FALSE, retry... nLoop: 6
06:13:26 CET: Available FALSE, retry... nLoop: 7
06:13:26 CET: Available FALSE, retry... nLoop: 8
06:13:27 CET: Available FALSE, retry... nLoop: 9
06:13:27 CET: Validation Path to oeserverd is: /home/mk/.var/app/org.opencpn.OpenCPN/bin/oeserverd
06:13:27 CET: Validation Path to server test is: /home/mk/.var/app/org.opencpn.OpenCPN/bin/oeserverd
06:13:27 CET: oesenc_pi: Checking oeserverd utility at {/home/mk/.var/app/org.opencpn.OpenCPN/bin/oeserverd}
06:13:27 CET: oesenc_pi: starting oeserverd utility: /home/mk/.var/app/org.opencpn.OpenCPN/bin/oeserverd
06:13:30 CET: oesenc_pi: oeserverd utility at
{/home/mk/.var/app/org.opencpn.OpenCPN/bin/oeserverd}
 reports Unavailable.

```

**Comment by leamas** [comment]
Filed oesenc PR: https://github.com/bdbcat/oesenc_pi/pull/82

**Comment by dominig** [comment]
There is an other way to follow fast developing process from Distro; It consists to use their integrated public build infra or a multi platform one.
I do generate package for multiple version of of OpenSUSE, including for Thumbleweed which changes several time per single day. But as the building work and update process is run by the distro system itself, it's almost hassle free (still have to deal with compilation errors when incompatibility comes with new packages).

Ubuntu provides it's PPA, https://launchpad.net/ubuntu/+ppas
Fedora has Koji https://koji.fedoraproject.org/koji/ 
OpenSUSE provides the OBS https://openbuildservice.org/

The particularity of OBS is that it can build for other distro than OpenSUSE.
I did provide for friends a few month ago, Fedora 32 releases with source rpm as OCPN official page provides only the binary.
If that is of use to anyone I could reactivate Fedora target. Please just let me know if there is a need.
In one way it would be simpler and better for Fedora user to build Fedora release with Koji. I could set up the initial build if there is an interest, but as I am not using Fedora, I am not the best position to decide what is suitable for released or not.

**Comment by leamas** [comment]
I don't see the point. And I'm saying that as a  Fedora user since many year, having no plans to change that. 

Your reasoning makes sense when it comes to the main opencpn package. But the plugins is another issue. The new plugin loader means that we distribute precompiled plugins for a number of platforms, for linux that is Debian and flatpak. These plugins cannot really be part of a distribution mechanism like COPR or a launchpad PPA for both technical and,  not least, legal reasons. Which means that we need to compile and distribute the plugins ourselves to be able to provide something similar to the plugin mechanism in for example browsers

And, we just don't have the manpower to handle more Linux platforms. The matrix [platforms X plugins] is daunting.

Not to forget how attractive the flatpak system actually is. It works, actually works, out of the box any any linux distro you can imagine, without added work. The downside is the package size, but in short I think it's worth it.

Bottom line: no problem creating more opencpn packages. But the plugins is something different, plugins which are required for a usable application

**Comment by leamas** [comment]
BTW: This one should be closed, the fedora plugins are dropped. So I close it. That said, please feel free to continue this discussion if you want.
